django/db/models/expressions.py

```diff
@@ -866,6 +895,112 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
         return sql, sql_params
 
 
+class SubQuery(Expression):
+    """
+    An explicit subquery. It may contain OuterRef() references to the outer
+    query, which will be resolved when it is applied to that query.
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery.all()
```

Comment:
I'm torn whether or not this copy is necessary. When we resolve the expression we do a copy of the subquery anyway. Even if the queryset was cached and evaluated, the resolving will copy a new queryset anyway.

```python
qs = Model.objects.whatever()
sq = SubQuery(qs)
list(qs) # this evaluates the queryset that subquery is holding onto
OtherModel.objects.annotate(subq=sq)  # queryset is copied here anyway, previous eval doesn't matter
```

Let me know if you can poke holes in my reasoning (it is new years day after all...).